editor/modes/visual-editor/block.js

```diff
@@ -264,6 +323,92 @@ class VisualEditorBlock extends Component {
 		this.removeOrDeselect( event );
 	}
 
+	focusToolbarItem( after, reverseOrder ) {
+		const block = findDOMNode( this.node );
+		const isVisible = ( elem ) => elem && ! ( elem.offsetWidth <= 0 || elem.offsetHeight <= 0 );
+		const allToolbars = filter( Array.from( block.querySelectorAll( '.components-toolbar' ) ), isVisible );
+		const settingsMenu = filter( [ block.querySelector( '.editor-block-settings-menu' ) ], isVisible );
+		const moverMenu = filter( [ block.querySelector( '.editor-block-mover' ) ], isVisible );
+		const allCycle = [ ...allToolbars, ...settingsMenu, ...moverMenu ];
+
+		if ( reverseOrder ) {
+			allCycle.reverse();
+		}
+
+		let startIndex = 0;
+		if ( after ) {
+			const currIndex = findIndex( allCycle, ( node ) => node.contains( after ) );
+			if ( currIndex === -1 ) {
+				return false;
+			}
+			startIndex = currIndex + 1;
+		}
+
+		// try all toolbars after this one to find the first with a focusable item
+		for ( let i = 0; i < allCycle.length; i++ ) {
+			const nextToolbar = allCycle[ ( startIndex + i ) % allCycle.length ];
+
+			const firstFocusableItem = nextToolbar.querySelector( '*[tabindex="-1"]:not(:disabled)' );
+
+			if ( firstFocusableItem && isVisible( firstFocusableItem ) ) {
+				firstFocusableItem.focus();
+				return true;
+			}
+		}
+		// nothing focusable
+		return false;
+	}
+
+	handleToolbarTabCycle( event ) {
+		const { keyCode, shiftKey, target } = event;
+		if ( keyCode !== TAB || ! this.node ) {
+			return;
+		}
+
+		if ( this.focusToolbarItem( target, shiftKey ) ) {
+			// we have a toolbar selected so we should be tabbing between toolbars
+			event.preventDefault();
+			event.stopPropagation();
+		}
+	}
+
+	handleToolbarArrowCycle( event ) {
+		const { keyCode, target } = event;
+		if ( ! ( keyCode === UP || keyCode === LEFT || keyCode === DOWN || keyCode === RIGHT ) || ! this.node ) {
+			return;
+		}
+		const forward = keyCode === DOWN || keyCode === RIGHT;
+		const block = findDOMNode( this.node );
```

Comment:
`this.node` should already be an HTMLElement, since it's assigned from the root `<div>`. You should only need `findDOMNode` if you're attaching a `ref` to another component, or when passing `this` as an argument (which would have the same result).